plugins/source/slack/client/rate_limiting.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"math/rand"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/slack-go/slack"
+)
+
+// RetryOnError will run the given resolver function and retry on rate limit exceeded errors
+// or other retryable errors (like internal server errors) after waiting some amount of time.
+func (c *Client) RetryOnError(tableName string, f func() error) error {
+	retries := 0
+	for err := f(); err != nil; err = f() {
+		switch typed := err.(type) {
+		case *slack.RateLimitedError:
+			randFloat64 := rand.Float64()
+			jitter := time.Duration(randFloat64 * float64(c.backoff))
+			retryAfter := typed.RetryAfter + jitter
+			c.logger.Info().Str("table", tableName).Msgf("Rate limit exceeded, retrying in %.2fs", retryAfter.Seconds())
+			time.Sleep(retryAfter)
+		default:
+			if !isRetryable(err) || retries >= c.maxRetries {
+				return err
+			}
+			randFloat64 := rand.Float64()
+			retryAfter := time.Duration(randFloat64 * float64(c.backoff))
+			retries++
+			c.logger.Info().Str("table", tableName).Msgf("Got retryable error (%v), retrying in %.2fs (%d/%d)", err, retryAfter.Seconds(), retries, c.maxRetries)
+			time.Sleep(retryAfter)
+		}
+	}
+	return nil
+}
+
+func isRetryable(err error) bool {
+	switch {
+	case strings.Contains(err.Error(), http.StatusText(http.StatusInternalServerError)):
```

Comment:
Not good form as the `StatusText` could be variable depending on server (it's not, but there's nothing preventing it to be?) so I would check if that's the case. Better way would be to unwrap until you get to a typed httperror and check response.Status int instead, but if the library doesn't do that (wrap the error response status codes in err) it would be just too complicated to add it on as an afterthought.

Reply:
The library doesn't expose the http error afaict, so this is a best-effort based retry based on a few errors I have actually observed where the message did match the status text

Reply:
Ok, I think I found a way to handle this better... will try updating